server.js

```diff
@@ -47,6 +47,228 @@ app.get('/authenticate', (request, response) => {
+      response.status(500).json({ error })
+    });
+});
+
+app.get('/api/v1/students', (request, response) => {
+  database('students').select()
+    .then(students => {
+      if (students) {
+        response.status(200).json(students)
+      }
+      return response.status(404).json({
+        error: 'There were no students found!'
+      })
+    })
+    .catch(error => {
+      response.status(500).json({ error })
+    });
+});
+
+app.get('/api/v1/programs', (request, response) => {
+  database('programs').select()
+    .then(programs => {
+      if (programs) {
+        response.status(200).json(programs)
+      }
+      return response.status(404).json({
+        error: 'There were no programs found!'
+      })
+    })
+    .catch(error => {
+      response.status(500).json({ error })
+    });
+});
+
+app.get('/api/v1/mentors/:id', (request, response) => {
+  const { id } = request.params;
+
+  database('mentors').where('id', id).select()
+    .then(mentor => {
+      if (mentor) {
+        response.status(200).json(mentor)
+      }
+      return response.status(404).json({
+        error: 'We could not find that mentor!'
+      })
+    })
+    .catch(error => {
+      response.status(500).json({ error })
+    });
+});
+
+app.get('/api/v1/students/:id', (request, response) => {
+  const { id } = request.params;
+
+  database('students').select()
+    .then(student => {
+      if (student) {
+        response.status(200).json(student)
+      }
+      return response.status(404).json({
+        error: 'We could not find that student!'
+      })
+    })
+    .catch(error => {
+      response.status(500).json({ error })
+    });
+});
+
+app.get('/api/v1/programs/:id', (request, response) => {
+  const { id } = request.params;
+
+  database('programs').select()
+    .then(program => {
+      if (program) {
+        response.status(200).json(program)
+      }
+      return response.status(404).json({
+        error: 'There were no program found!'
+      })
+    })
+    .catch(error => {
+      response.status(500).json({ error })
+    });
+});
+
+app.post('/api/v1/mentors', (request, response) => {
+
+  const mentor = request.body;
+
+  database('mentors').insert(mentor, 'id')
+    .then(mentor => {
+      if (result) {
+        resonse.status(201).json({ id: mentor[0] });
+      } else {
+        response.status(201).json({ error: 'You are missing a property'});
+      }
+    })
+    .catch(error => {
+      response.status(500).json({ error });
+    });
+});
+
+app.post('/api/v1/students', (request, response) => {
+
+  const student = request.body;
+
+  database('students').insert(student, 'id')
+    .then(student => {
+      if (result) {
+        resonse.status(201).json({ id: student[0] });
+      } else {
+        response.status(201).json({ error: 'You are missing a property'});
+      }
+    })
+    .catch(error => {
+      response.status(500).json({ error });
+    });
+});
+
+app.post('/api/v1/programs', (request, response) => {
```

Comment:
I think we can leave this one out. It'd be easy enough for future people to add, if Turing adds programs.